### PR TITLE
Offline cert renewal

### PR DESCRIPTION
**Pull request by SilleBille**
This patch adds the `cert-fix` module which can be used to renew
all expired certificates. The `cert-fix` module is built as a high level
CLI on top of existing `cert` module components.

Ticket: https://pagure.io/dogtagpki/issue/2776

Signed-off-by: Dinesh Prasanth M K <dmoluguw@redhat.com>

### CODE REVIEW

File: base/server/python/pki/server/__init__.py [hunk 1]
```diff
@@ -1005,6 +1005,79 @@ def __repr__(self):
...
+            cmd.extend(['selftest-enable'])
+        else:
+            cmd.extend(['selftest-disable'])
+
+        if subsystem:
+            cmd.extend(['--subsystem', subsystem])
+
+        cmd.extend(['--instance', self.name])
+
+        subprocess.check_call(cmd)
```
Comment (edewata) [inline base/server/python/pki/server/__init__.py]: I think we should avoid going from Python to Shell then to Python again. If the code that we want can be called directly from Python that would be better. So you might want to refactor      the code out of selftest.py first.

File: base/server/python/pki/server/__init__.py [hunk 2]
```diff
@@ -1005,6 +1005,79 @@ def __repr__(self):
...
+
+        if temp:
+            cmd.extend(['--temp'])
+
+        if renew:
+            cmd.extend(['--renew'])
+
+        logger.info('Executing CMD: %s', cmd)
+
+        subprocess.check_call(cmd)
```
Comment (edewata) [inline base/server/python/pki/server/__init__.py]: Same thing here, refactor the code out of cert.py first.

File: base/server/python/pki/server/__init__.py [hunk 3]
```diff
@@ -1005,6 +1005,79 @@ def __repr__(self):
...
+    def cert_del(self, cert_id):
+
+        cmd = [
+            'pki-server', 'cert-del', '--instance', self.name
+        ]
+        if cert_id:
+            cmd.extend([cert_id])
+
+        logger.info('Executing CMD: %s', cmd)
+        subprocess.check_call(cmd)
```
Comment (edewata) [inline base/server/python/pki/server/__init__.py]: And here too.

**Comment by SilleBille** [comment]
This patch has become too huge to review. I'll be submitting a series of patches and will be abandoning this patch.

**Comment by SilleBille** [comment]
Closing this PR since it has been merged via a different PR: #98  and the related refactoring have done through different PRs.

**Pull request closed without merging**